packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -50,8 +54,12 @@ export class ApplicationExtension<
      * @param routes - The list application routes currently loaded.
      * @returns routes - The modified application routes.
      */
-    public extendRoutes(routes: RouteProps[]): RouteProps[] {
-        return routes
+    public extendRoutes(routes: RouteProps[]): Promise<RouteProps[]> {
+        return Promise.resolve(routes)
+    }
+
+    public getRoutes(): Promise<RouteProps[]> {
```

Comment:
I'm all for this change to `getRoutes` excluding the `routes` parameter. Lets move forward with that.